src/components/Maps/MapPage.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { useQuery } from 'react-apollo';
+import GoogleMapReact from 'google-map-react';
+
+import MenuAppBar from '../common/MenuAppBar';
+import InfoWindow from './InfoWindow';
+import { LocationPin, GET_PINS_BY_EVENT_ID } from '../../graphql/queries/maps';
+
+const MapPage = ({
+  match: {
+    params: { eventId },
+  },
+}: {
+  match: { params: { eventId: string } };
+}) => {
+  const defaultMap = {
+    center: { lat: 43.470846, lng: -80.538473 },
+    zoom: 11,
+  };
+
+  const getMapOptions = (maps) => {
+    return {
+      streetViewControl: false,
+      scaleControl: true,
+      fullscreenControl: false,
+      styles: [
+        {
+          featureType: 'poi.business',
+          elementType: 'labels',
+          stylers: [
+            {
+              visibility: 'off',
+            },
+          ],
+        },
+      ],
+      gestureHandling: 'greedy',
+      disableDoubleClickZoom: true,
+
+      mapTypeControl: true,
+      mapTypeId: maps.MapTypeId.ROADMAP,
+      mapTypeControlOptions: {
+        style: maps.MapTypeControlStyle.HORIZONTAL_BAR,
+        position: maps.ControlPosition.BOTTOM_CENTER,
+        mapTypeIds: [
+          maps.MapTypeId.ROADMAP,
+          maps.MapTypeId.SATELLITE,
+          maps.MapTypeId.HYBRID,
+        ],
+      },
+
+      zoomControl: true,
+      clickableIcons: false,
+    };
+  };
+
+  const { data, loading } = useQuery(GET_PINS_BY_EVENT_ID, {
+    variables: { eventId },
+  });
+
+  const pins: Array<LocationPin> = data && !loading ? data.pinsForEvent : [];
+  const [infoWindowOpen, setInfoWindowOpen] = React.useState(false);
+  const [interestPinTitle, setInterestPinTitle] = React.useState('');
+  const [interestPinLocation, setInterestPinLocation] = React.useState('');
+
+  const initMaps = (map, maps) => {
+    if (!loading) {
+      map.addListener('click', () => {
+        setInfoWindowOpen(false);
+        setInterestPinLocation('');
+        setInterestPinTitle('');
+      });
+      pins.map((pin) => {
+        const marker = new maps.Marker({
+          position: { lat: pin.latitude, lng: pin.longitude },
+          map,
+          title: pin.label,
+        });
+
+        marker.addListener('click', () => {
+          map.setCenter(marker.getPosition());
+          setInfoWindowOpen(true);
+          setInterestPinLocation(pin.address);
+          setInterestPinTitle(pin.label);
+        });
+
+        return marker;
+      });
+
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition((position: Position) => {
+          const pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude,
+          };
+
+          const marker = new maps.Marker({
+            position: pos,
+            map,
+            title: 'Current position',
+          });
+
+          marker.setIcon(
+            'http://maps.google.com/mapfiles/ms/icons/green-dot.png'
```

Comment:
does this make a network request from google maps? is it possible to just use a `material-ui` asset/icon instead?

Reply:
We can make all our markers into `material-ui` icons but the issue is that we cannot link map functionality with the marker. For example, we currently make the map center on the pin that is clicked. If we were to use `material-ui` icons, we cannot use the `initMaps` function, thus blocking us from setting any sort of map functionality.

Reply:
actually forgot to mention before that the centering of the pin experience isn't one that me or rishabh like - it's very jarring to the user when selecting pins and can lead to users losing the context of where they were on the map before they selected a pin

is it possible to go back to the pattern of changing the colour of the selected pin instead of shifting the map to centre it?